diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7bcb6ba..50c9536 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,8 +30,10 @@ endif()
 
 # define executable to build
 include_directories("inc")
+link_libraries("-static")
 add_executable(${PROJECT_NAME} main.cpp)
 
+
 if (COVERAGE)
 	set(COVERAGE_FLAGS "-g -O0 -fprofile-arcs -ftest-coverage")
 	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${COVERAGE_FLAGS}")
diff --git a/inc/BaseServerMediaSubsession.h b/inc/BaseServerMediaSubsession.h
index 1ab3d8d..d657c59 100755
--- a/inc/BaseServerMediaSubsession.h
+++ b/inc/BaseServerMediaSubsession.h
@@ -59,7 +59,6 @@ class BaseServerMediaSubsession
             switch(format)
             {	
                 case V4L2_PIX_FMT_HEVC  : rtpFormat = "video/H265"; break;
-                case V4L2_PIX_FMT_H264  : rtpFormat = "video/H264"; break;
                 case V4L2_PIX_FMT_MJPEG : rtpFormat = "video/JPEG"; break;
                 case V4L2_PIX_FMT_JPEG  : rtpFormat = "video/JPEG"; break;
                 case V4L2_PIX_FMT_VP8   : rtpFormat = "video/VP8" ; break;
diff --git a/inc/DeviceSourceFactory.h b/inc/DeviceSourceFactory.h
index 093c1f3..16bcb46 100644
--- a/inc/DeviceSourceFactory.h
+++ b/inc/DeviceSourceFactory.h
@@ -21,11 +21,7 @@ class DeviceSourceFactory {
     public:
 		static FramedSource* createFramedSource(UsageEnvironment* env, int format, DeviceInterface* devCapture, int queueSize = 5, V4L2DeviceSource::CaptureMode captureMode = V4L2DeviceSource::CAPTURE_INTERNAL_THREAD, int outfd = -1, bool repeatConfig = true) {
             FramedSource* source = NULL;
-            if (format == V4L2_PIX_FMT_H264)
-            {
-                source = H264_V4L2DeviceSource::createNew(*env, devCapture, outfd, queueSize, captureMode, repeatConfig, false);
-            }
-            else if (format == V4L2_PIX_FMT_HEVC)
+            if (format == V4L2_PIX_FMT_HEVC)
             {
                 source = H265_V4L2DeviceSource::createNew(*env, devCapture, outfd, queueSize, captureMode, repeatConfig, false);
             }
diff --git a/main.cpp b/main.cpp
index 9581258..aacb164 100755
--- a/main.cpp
+++ b/main.cpp
@@ -250,7 +250,6 @@ int main(int argc, char** argv)
 	
 	// default format tries
 	if ((videoformatList.empty()) && (format!=0)) {
-		videoformatList.push_back(V4L2_PIX_FMT_H264);
 		videoformatList.push_back(V4L2_PIX_FMT_MJPEG);
 		videoformatList.push_back(V4L2_PIX_FMT_JPEG);
 		videoformatList.push_back(V4L2_PIX_FMT_NV12);
